app/state/runtime.js

```diff
@@ -48,30 +48,52 @@ export const reducer = (previousState = initialState, action) => {
       const ast = parser(previousState.get("source"));
       const kernel = compile(ast);
       const runtime = buildFromKernelAST(kernel);
-      const step = executeSingleStep(runtime);
-      return previousState
-        .set("steps", Immutable.List([step]))
-        .set("currentStep", 0);
+      try {
+        const step = executeSingleStep(runtime);
+        return previousState
+          .set("steps", Immutable.List([step]))
```

Comment:
You can use `Immutable.List.of(step)` instead of manually creating a list with only one element.